src/api/utils/api_utils.py

```diff
@@ -327,3 +335,82 @@ def get_supported_language(language_code):
     if supported_language:
         return supported_language.code
     return DEFAULT_SOURCE_LANGUAGE_CODE
+
+
+# ------------------------ Recurring Events utils ----------------------------------------#
+
+def get_final_date(event, today):
+    start_time = event.start_date_and_time.strftime("%H:%M:%S+00:00")
+    final_date_str = event.recurring_details.get('final_date')
+    
+    if final_date_str and final_date_str != 'None':
+        final_date = datetime.strptime(f"{final_date_str} {start_time}", "%Y-%m-%d %H:%M:%S+00:00")
+        return final_date.replace(tzinfo=custom_timezone_info())
```

Comment:
nit: in general i think these helper functions can actually live in the events file themselves.  except that you want to use it in different places?

Reply:
For now, the functions are used in only one place. I will move them to the events file.